src/overlays/actors/ovl_En_Go2/z_en_go2.c

```diff
@@ -811,15 +842,24 @@ s16 EnGo2_UpdateTalkState(PlayState* play, Actor* thisx) {
 #endif
 }
 
-s32 func_80A44790(EnGo2* this, PlayState* play) {
-    if (PARAMS_GET_S(this->actor.params, 0, 5) != GORON_DMT_BIGGORON &&
-        PARAMS_GET_S(this->actor.params, 0, 5) != GORON_CITY_ROLLING_BIG) {
-        return Npc_UpdateTalking(play, &this->actor, &this->interactInfo.talkState, this->interactRange,
-                                 EnGo2_GetTextId, EnGo2_UpdateTalkState);
-    } else if ((PARAMS_GET_S(this->actor.params, 0, 5) == GORON_DMT_BIGGORON) &&
-               !(this->collider.base.ocFlags2 & OC2_HIT_PLAYER)) {
-        return false;
-    } else {
+s32 EnGo2_UpdateTalking(EnGo2* this, PlayState* play) {
+    // default:
+    if (ENGO2_GET_TYPE(this) != GORON_DMT_BIGGORON) {
+        if (ENGO2_GET_TYPE(this) != GORON_CITY_HOT_RODDER) {
+            return Npc_UpdateTalking(play, &this->actor, &this->interactInfo.talkState, this->interactRange,
+                                     EnGo2_GetTextId, EnGo2_UpdateTalkState);
+        }
+    }
+
+    // `GORON_DMT_BIGGORON`, close enough; see `EnGo2_IsInRange`
+    if (ENGO2_GET_TYPE(this) == GORON_DMT_BIGGORON) {
+        if (!(this->collider.base.ocFlags2 & OC2_HIT_PLAYER)) {
+            return false;
+        }
+    }
+
+    // `GORON_DMT_BIGGORON` || `GORON_CITY_HOT_RODDER`
+    {
```

Comment:
```suggestion
    } else {
```
I'd return the else statement here, and hoist the ``GORON_DMT_BIGGORON` || `GORON_CITY_HOT_RODDER`` comment up so it's before the `ENGO2_GET_TYPE(this) == GORON_DMT_BIGGORON` check. I'd also just write the comment out in english over using `||`